### PR TITLE
[arrow] add cuda to feature list

### LINKED ISSUE #24465: [arrow] Request for cuda support
Can `cuda` please be added to the supported packages available for arrow?
As of release 2022.04.12 they are:
```
$ ./vcpkg.exe search arrow
arrow                    7.0.0#2          Cross-language development platform for in-memory analytics
arrow[csv]                                CSV support
arrow[dataset]                            Dataset support
arrow[filesystem]                         Filesystem support
arrow[flight]                             Arrow Flight RPC support
arrow[jemalloc]                           jemalloc allocator
arrow[json]                               JSON support
arrow[mimalloc]                           mimalloc allocator
arrow[orc]                                ORC support
arrow[parquet]                            Parquet support
arrow[plasma]                             Plasma support
arrow[s3]                                 S3 support
```

### PR DESCRIPTION
**Pull request by Adela0814**
Fix #24465

Add `cuda `to feature list

tested features in `linux`

### CODE REVIEW

**Review by JonLiu1993 — APPROVED** [review]

File: ports/arrow/vcpkg.json [hunk 1]
```diff
@@ -41,6 +41,13 @@
     "csv": {
       "description": "CSV support"
     },
+    "cuda": {
+      "description": "cuda support",
+      "supports": "linux | (!osx & !uwp & !(arm64 & windows))",
+      "dependencies": [
+        "cuda"
```
Comment (BillyONeal) [inline ports/arrow/vcpkg.json:48]: This already has:

`"supports": "(windows & x64 & !uwp) | (linux & x64) | (linux & arm64)"`

Do we need to effectively duplicate the condition here?

(Also can we not have 3 level deep nesting and negation?)

**Review by BillyONeal — CHANGES_REQUESTED** [review]
I would also like to see @mattpulver 's comment addressed.

File: ports/arrow/vcpkg.json (hunk 1 above)
Comment (Adela0814) [reply ports/arrow/vcpkg.json:48]: Maybe it should be changed to `(windows & x64 & !uwp) | linux`, do we need to consider `linux-arm64`? this is a community triplet.

Comment (BillyONeal) [reply ports/arrow/vcpkg.json:48]: I'm saying that it seems to be just forwarding cuda's requirements so maybe it shouldn't have a "supports" at all?

Comment (Adela0814) [reply ports/arrow/vcpkg.json:47]: Yes, you are right, I have removed it. thanks for your comment.

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Comment by mattpulver** [comment]
Thank you for your prompt attention to this feature request.

I tried building w/ `master` commit 0a1420dc443a2c8fc135ef88bbf96af57263e911 but am still encountering the compiler error
```
fatal error C1083: Cannot open include file: 'arrow/gpu/cuda_api.h': No such file or directory
```
To reproduce this, one need only compile a project with the line
```
#include <arrow/gpu/cuda_api.h>
```
My expectation is that arrow/cuda support would allow for this to compile, as it does in Linux without vcpkg support. If this expectation is incorrect, what is the equivalent?

**Comment by BillyONeal** [comment]
> My expectation is that arrow/cuda support would allow for this to compile, as it does in Linux without vcpkg support. If this expectation is incorrect, what is the equivalent?

This PR just hooked up the build option; it doesn't look like we checked that the upstream build system necessarily installs the right header :)

**Comment by mattpulver** [comment]
> This PR just hooked up the build option; it doesn't look like we checked that the upstream build system necessarily installs the right header :)

Submitted as new issue: #24678

**Pull request merged**